diff -urN bz2.orig\bzlib_pr.h bz2\bzlib_pr.h
--- bz2.orig\bzlib_pr.h	Mon Mar 11 20:32:51 2013
+++ bz2\bzlib_pr.h	Mon Mar 11 20:32:51 2013
@@ -85,7 +85,13 @@
 
 #else
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern void bz_internal_error ( int errcode );
+#ifdef __cplusplus
+}
+#endif
 #define AssertH(cond,errcode) \
    { if (!(cond)) bz_internal_error ( errcode ); }
 #define AssertD(cond,msg)                do { } while (0)
