diff --strip-trailing-cr -urN minilzo.orig/lzodefs.h minilzo/lzodefs.h
--- minilzo.orig/lzodefs.h	2011-08-12 16:02:59.000000000 +0000
+++ minilzo/lzodefs.h	2024-12-11 09:52:14.000000000 +0000
@@ -670,6 +670,9 @@
 #  define LZO_ARCH_I086             1
 #  define LZO_ARCH_IA16             1
 #  define LZO_INFO_ARCH             "i086"
+#elif defined(__aarch64__) || defined(_M_ARM64)
+#  define LZO_ARCH_ARM64            1
+#  define LZO_INFO_ARCH             "arm64"
 #elif defined(__alpha__) || defined(__alpha) || defined(_M_ALPHA)
 #  define LZO_ARCH_ALPHA            1
 #  define LZO_INFO_ARCH             "alpha"
diff --strip-trailing-cr -urN minilzo.orig/minilzo.c minilzo/minilzo.c
--- minilzo.orig/minilzo.c	2011-08-12 16:02:59.000000000 +0000
+++ minilzo/minilzo.c	2025-09-09 13:42:08.000000000 +0000
@@ -59,6 +59,11 @@
 #endif
 #include <limits.h>
 #include <stddef.h>
+#if defined(__cplusplus) && __cplusplus>=201703L
+#  define LZO_REGISTER
+#else
+#  define LZO_REGISTER register
+#endif
 #if defined(MINILZO_CFG_USE_INTERNAL_LZODEFS)
 
 #ifndef __LZODEFS_H_INCLUDED
@@ -3168,7 +3173,7 @@
                     lzo_bytep out, lzo_uintp out_len,
                     lzo_uint  ti,  lzo_voidp wrkmem)
 {
-    register const lzo_bytep ip;
+    LZO_REGISTER const lzo_bytep ip;
     lzo_bytep op;
     const lzo_bytep const in_end = in + in_len;
     const lzo_bytep const ip_end = in + in_len - 20;
@@ -3241,7 +3246,7 @@
 
         ii -= ti; ti = 0;
         {
-        register lzo_uint t = pd(ip,ii);
+        LZO_REGISTER lzo_uint t = pd(ip,ii);
         if (t != 0)
         {
             if (t <= 3)
@@ -3276,7 +3281,7 @@
                     *op++ = LZO_BYTE(t - 3);
                 else
                 {
-                    register lzo_uint tt = t - 18;
+                    LZO_REGISTER lzo_uint tt = t - 18;
                     *op++ = 0;
                     while __lzo_unlikely(tt > 255)
                     {
@@ -3623,14 +3628,14 @@
                        lzo_voidp wrkmem )
 #endif
 {
-    register lzo_bytep op;
-    register const lzo_bytep ip;
-    register lzo_uint t;
+    LZO_REGISTER lzo_bytep op;
+    LZO_REGISTER const lzo_bytep ip;
+    LZO_REGISTER lzo_uint t;
 #if defined(COPY_DICT)
     lzo_uint m_off;
     const lzo_bytep dict_end;
 #else
-    register const lzo_bytep m_pos;
+    LZO_REGISTER const lzo_bytep m_pos;
 #endif
 
     const lzo_bytep const ip_end = in + in_len;
@@ -4149,14 +4154,14 @@
                        lzo_voidp wrkmem )
 #endif
 {
-    register lzo_bytep op;
-    register const lzo_bytep ip;
-    register lzo_uint t;
+    LZO_REGISTER lzo_bytep op;
+    LZO_REGISTER const lzo_bytep ip;
+    LZO_REGISTER lzo_uint t;
 #if defined(COPY_DICT)
     lzo_uint m_off;
     const lzo_bytep dict_end;
 #else
-    register const lzo_bytep m_pos;
+    LZO_REGISTER const lzo_bytep m_pos;
 #endif
 
     const lzo_bytep const ip_end = in + in_len;
